### PR TITLE
ios_l2_interface

### PR DESCRIPTION
**Pull request by Qalthos**
##### SUMMARY

New module ios_l2_interface

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

ios_l2_interface

##### ANSIBLE VERSION

```
2.5
```

### CODE REVIEW

File: test/integration/targets/ios_l2_interface/tests/cli/agg.yaml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+- debug: msg="START connection={{ ansible_connection }} ios_l2_interface aggregate test"
+
+- name: Setup - Ensure interfaces are switchport
+  ios_config:
+    lines:
+      - switchport
+    parents:
+      - "interface {{ item }}"
+  with_items:
```
Comment (trishnaguha) [inline test/integration/targets/ios_l2_interface/tests/cli/agg.yaml]: Use `loop` instead?

File: test/integration/targets/ios_l2_interface/tests/cli/agg.yaml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      - vlan_id: 6
+      - vlan_id: 15
+    provider: "{{ cli }}"
+
+- name: Setup - Remove interface aggregate before testing
+  ios_l2_interface:
+    aggregate:
+      - { name: "{{ test_interface }}", mode: access, access_vlan: 6 }
+      - { name: "{{ test_interface2 }}", mode: access, access_vlan: 15 }
+    state: absent
```
Comment (trishnaguha) [inline test/integration/targets/ios_l2_interface/tests/cli/agg.yaml:26]: Do we not need `authorize` in the testcases for connection `local`?

Comment (trishnaguha) [reply test/integration/targets/ios_l2_interface/tests/cli/agg.yaml:26]: Oh never mind, just checked that it is ported to provider.

File: test/integration/targets/ios_l2_interface/tests/cli/agg.yaml (hunk 1 above)
Comment (Qalthos) [reply test/integration/targets/ios_l2_interface/tests/cli/agg.yaml]: That's probably a good idea

**Review by trishnaguha — APPROVED** [review]

File: lib/ansible/modules/network/ios/ios_l2_interface.py [hunk 3]
```diff
@@ -0,0 +1,488 @@
...
+  native_vlan:
+    description:
+      - Native VLAN to be configured in trunk port.
+        If C(mode=trunk), used as the trunk native VLAN ID.
+  trunk_allowed_vlans:
+    description:
+      - List of allowed VLANs in a given trunk port.
+        If C(mode=trunk), these are the only VLANs that will be
+        configured on the trunk, i.e. "2-10,15".
+  aggregate:
```
Comment (gundalow) [inline lib/ansible/modules/network/ios/ios_l2_interface.py:56]: Worth giving an example of `aggregate`?

**Pull request merged**